android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -415,22 +428,30 @@ public View getView(int i, View view, ViewGroup viewGroup) {
       setText(view, R.id.description, pwsResult.getDescription());
       ((ImageView) view.findViewById(R.id.icon)).setImageBitmap(
           Utils.getBitmapIcon(mPwCollection, pwsResult));
+      final String siteUrl = pwsResult.getSiteUrl();
+        if (Utils.isFavorite(pwsResult.getSiteUrl())) {
+          ((ImageView) view.findViewById(R.id.star)).setImageResource(android.R.drawable.btn_star_big_on);
+        } else {
+          ((ImageView) view.findViewById(R.id.star)).setImageResource(0);
```

Comment:
I think it would be cleaner to turn the favorites icon on and off using setVisibility rather than changing the image resource.